extension/duckdb_scanner/src/duckdb_storage.cpp

```diff
@@ -1,11 +1,82 @@
 #include "duckdb_storage.h"
 
+#include "binder/ddl/bound_create_table_info.h"
+#include "catalog/catalog_entry/table_catalog_entry.h"
+#include "common/exception/binder.h"
+#include "duckdb_catalog.h"
 #include "duckdb_scan.h"
+#include "duckdb_type_converter.h"
 
 namespace kuzu {
 namespace duckdb_scanner {
 
-std::unique_ptr<main::AttachedDatabase> attachDuckDB(std::string dbName, std::string dbPath) {
+static void getTableInfo(duckdb::Connection& con, std::string tableName,
+    std::vector<common::LogicalType>& columnTypes, std::vector<std::string>& columnNames) {
+    auto result = con.Query(
+        common::stringFormat("select data_type,column_name from information_schema.columns where "
+                             "table_name = '{}' and table_schema='main';",
+            tableName));
+    if (result->RowCount() == 0) {
+        throw common::BinderException(
```

Comment:
This exception is caught in the caller but never throw out. So I guess you will never be able to throw this exception to the user end. So you might wanna just return a flag.